### PR TITLE
Docker Improvements

### PR DESCRIPTION
**Pull request by gartens**
## Summary

Various improvements to the Docker subsystem, such as:
 - Getting rid of Map.of in favor of properly annotated records/classes
 - Handshakes are now identified by an id, not the hostname
 - Better error handling by introducing `DockerUserException` which carries an appropriate HTTP status code for the response.
 - Change the schema of the endpoints to contain the ids, be a bit more CRUD like and use HTTP verbs like `DELETE` etc.
 - Various other small improvements

See [#94](https://github.com/polypheny/Polypheny-UI/pull/94) for the UI changes.

### CODE REVIEW

File: core/src/main/java/org/polypheny/db/docker/DockerInstance.java [hunk 1]
```diff
@@ -269,24 +271,39 @@ boolean hasContainers() throws IOException {
...
+//
+//                    }
+//                    // Getting here means everything is OK.
+//                } catch ( IOException e ) {
+//                    log.error( "Failed to use new Docker instance", e );
+//                    throw new DockerUserException( "Cannot update to Docker instance" );
+//                }
+//
+                // TODO: Copy/Move keys...
+                // TODO: Restart all proxy connections
```
Comment (datomo) [inline core/src/main/java/org/polypheny/db/docker/DockerInstance.java:295]: Open todo

File: core/src/main/java/org/polypheny/db/docker/DockerInstance.java [hunk 2]
```diff
@@ -269,24 +271,39 @@ boolean hasContainers() throws IOException {
...
+//                    client.ping();
+//                    if ( !client.getDockerId().equals( dockerInstanceUuid ) ) {
+//
+//                    }
+//                    // Getting here means everything is OK.
+//                } catch ( IOException e ) {
+//                    log.error( "Failed to use new Docker instance", e );
+//                    throw new DockerUserException( "Cannot update to Docker instance" );
+//                }
+//
```
Comment (datomo) [inline core/src/main/java/org/polypheny/db/docker/DockerInstance.java:293]: Seems like leftover artifact.

File: core/src/main/java/org/polypheny/db/docker/PolyphenyHandshakeClient.java [hunk 3]
```diff
@@ -143,7 +143,7 @@ private byte[] computeAuthenticationValueV1( byte[] psk, byte[] uuid, byte[] cha
 
 
     void prepareNextTry() {
-        // TOOD: merge into doHandshake somehow
+        // TODO: merge into doHandshake somehow
```
Comment (datomo) [inline core/src/main/java/org/polypheny/db/docker/PolyphenyHandshakeClient.java:146]: Leftover todo

File: core/src/main/java/org/polypheny/db/docker/models/AutoDockerResult.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.polypheny.db.docker.models;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.List;
+
+public record AutoDockerResult(@JsonProperty AutoDockerStatus status, @JsonProperty List<DockerInstanceInfo> instances) {
```
Comment (datomo) [inline core/src/main/java/org/polypheny/db/docker/models/AutoDockerResult.java:22]: Please format with codestyle.

File: core/src/main/java/org/polypheny/db/docker/models/AutoDockerStatus.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.polypheny.db.docker.models;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public record AutoDockerStatus(@JsonProperty boolean available, @JsonProperty boolean connected, @JsonProperty boolean running, @JsonProperty String status) {
```
Comment (datomo) [inline core/src/main/java/org/polypheny/db/docker/models/AutoDockerStatus.java:21]: Please format with codestyle.

**Review by datomo — CHANGES_REQUESTED** [review]
Thank you @gartens, this looks very nice and clean.
With the minor things and one final reformat out of the way this should be good to go.

File: core/src/main/java/org/polypheny/db/docker/DockerInstance.java (hunk 1 above)
Comment (gartens) [reply core/src/main/java/org/polypheny/db/docker/DockerInstance.java:281]: This is an opportunity for future improvements

**Review by vogti — APPROVED** [review]
Thank you, @gartens, for this PR!

**Review by datomo — APPROVED** [review]

**Pull request merged**